File: src/main/java/com/github/felixgail/musicbot/gplaymusic/StreamQualitySerializer.java
package com.github.felixgail.musicbot.gplaymusic;

import com.github.felixgail.gplaymusic.model.enums.StreamQuality;
import net.bjoernpetersen.musicbot.api.config.ConfigSerializer;
import net.bjoernpetersen.musicbot.api.config.SerializationException;
import org.jetbrains.annotations.NotNull;

final class StreamQualitySerializer implements ConfigSerializer<StreamQuality> {

    @Override
    public StreamQuality deserialize(@NotNull String s) throws SerializationException {
        try {
            return StreamQuality.valueOf(s);
        } catch (IllegalArgumentException e) {
            throw new SerializationException();
        }
    }

    @NotNull
    @Override
    public String serialize(StreamQuality streamQuality) {
        return streamQuality.name();
    }
}

File: src/main/java/com/github/felixgail/musicbot/gplaymusic/StreamQualitySerializer.kt
package com.github.felixgail.musicbot.gplaymusic

import com.github.felixgail.gplaymusic.model.enums.StreamQuality
import net.bjoernpetersen.musicbot.api.config.ConfigSerializer
import net.bjoernpetersen.musicbot.api.config.SerializationException

internal class StreamQualitySerializer : ConfigSerializer<StreamQuality> {
    @Throws(SerializationException::class)
    override fun deserialize(string: String): StreamQuality {
        try {
            return StreamQuality.valueOf(string)
        } catch (e: IllegalArgumentException) {
            throw SerializationException()
        }
    }

    override fun serialize(obj: StreamQuality): String {
        return obj.name
    }
}
